sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/PartitionedQueryExecutionInfo.java

```diff
@@ -63,6 +65,17 @@ public RequestTimeline getQueryPlanRequestTimeline() {
         return queryPlanRequestTimeline;
     }
 
+    public boolean hasHybridSearchQueryInfo() {
+        getHybridSearchQueryInfo();
+        return hybridSearchQueryInfo != null;
+    }
+
+    public HybridSearchQueryInfo getHybridSearchQueryInfo() {
+        return this.hybridSearchQueryInfo != null ? this.hybridSearchQueryInfo
```

Comment:
no side effects in getters

Reply:
Ditto as above: Updated the code the code to use lazy initialization, which only happens during the first access.